lte/cloud/go/services/subscriberdb/servicers/subscriberdb_servicer.go

```diff
@@ -74,202 +106,71 @@ func (s *subscriberdbServicer) ListSubscribers(ctx context.Context, req *lte_pro
+	if !s.flatDigestEnabled {
+		return &lte_protos.SubscribersDigest{Md5Base64Digest: ""}, false
 	}
 
-	digest, err := getDigest(networkID, apnsByName, apnResourcesByAPN)
+	digest, err := s.getDigest(networkID)
 	// If digest generation fails, the error is swallowed to not affect the main functionality
 	if err != nil {
-		glog.Errorf("Generating digest for subscribers in network of gateway %s failed", networkID)
-	} else {
-		digestProto = &lte_protos.SubscribersDigest{Md5Base64Digest: digest}
-		noUpdates = req.Digest.Md5Base64Digest == digest
+		glog.Errorf("Generating digest for network %s failed: %+v", networkID, err)
+		return &lte_protos.SubscribersDigest{Md5Base64Digest: ""}, false
 	}
-	return digestProto, noUpdates
-}
 
-// getDigest returns a deterministic digest of all subscribers in the network.
-// Digest is generated by loading all subscribers in the network.
-func getDigest(
-	networkID string,
-	apnsByName map[string]*lte_models.ApnConfiguration,
-	apnResourcesByAPN lte_models.ApnResources,
-) (string, error) {
-	subProtosById := map[string]proto.Message{}
-	curToken := ""
-	foundEmptyToken := false
-	for !foundEmptyToken {
-		subProtos, nextToken, err := loadSubProtosPage(0, curToken, networkID, apnsByName, apnResourcesByAPN)
-		if err != nil {
-			return "", err
-		}
-		for _, subProto := range subProtos {
-			subProtosById[subProto.Sid.Id] = subProto
-		}
-		foundEmptyToken = nextToken == ""
-		curToken = nextToken
-	}
-	return mproto.HashManyDeterministic(subProtosById)
+	noUpdates := digest != "" && clientDigest != nil && clientDigest.Md5Base64Digest == digest
+	digestProto := &lte_protos.SubscribersDigest{Md5Base64Digest: digest}
+	return digestProto, noUpdates
 }
 
-func loadSubProtosPage(
-	pageSize uint32, pageToken string, networkID string,
-	apnsByName map[string]*lte_models.ApnConfiguration,
-	apnResourcesByAPN lte_models.ApnResources,
-) ([]*lte_protos.SubscriberData, string, error) {
-
-	lc := configurator.EntityLoadCriteria{
-		PageSize:           pageSize,
-		PageToken:          pageToken,
-		LoadConfig:         true,
-		LoadAssocsToThis:   true,
-		LoadAssocsFromThis: true,
-	}
-
-	subEnts, nextToken, err := configurator.LoadAllEntitiesOfType(
-		networkID, lte.SubscriberEntityType, lc, serdes.Entity,
-	)
-	if err != nil {
-		return nil, "", errors.Wrapf(err, "load subscribers in network of gateway %s", networkID)
-	}
-
-	subProtos := make([]*lte_protos.SubscriberData, 0, len(subEnts))
-	for _, sub := range subEnts {
-		subProto, err := convertSubEntsToProtos(sub, apnsByName, apnResourcesByAPN)
-		if err != nil {
-			return nil, "", err
-		}
-		subProto.NetworkId = &protos.NetworkID{Id: networkID}
-		subProtos = append(subProtos, subProto)
+func (s *subscriberdbServicer) getDigest(networkID string) (string, error) {
+	// The functionality of reading digests from the digest store is currently
+	// placed behind a feature flag
+	var digest string
+	var err error
+	if s.useDigestStore {
```

Comment:
Since we're doing this all in one go now: we really only have two options right now
- (a) Send digest down, and we always want to use the digest store for that
- (b) Don't send digest down at all, in which case we can just always send down empty string